### DISCUSSION TITLE
Displaying SVG in the lightbox, image is very small

### DISCUSSION THREAD

**Discussion by JPVRS** [igordanchenko/yet-another-react-lightbox#220] [Q&A] [answered]
As the title says, I have an SVG image that I'm trying to display in the lightbox. The image is sized pretty small and the zoom doesn't seem to be working. Any idea how I can get this to display like a regular image?

**Comment by igordanchenko** [comment]
I'm pretty sure the image is sized according to the `width` and `height` specified in the `svg` file. Zoom is probably disabled because the image is already at its "max zoom" level according to the `maxZoomPixelRatio` value (which is set to 1 [by default](https://yet-another-react-lightbox.com/plugins/zoom#Documentation)). I'm just making a guess here, as there is no demo to troubleshoot... 

You could probably increase `maxZoomPixelRatio` value, but that would affect all of your slides which isn't great. 

What are your expectations of the zoom feature in this case? How would you determine reasonable "max zoom" level for `svg` files?  How would you expect the lightbox to display `svg` images by default (at the initial zoom level)?

**Comment by JPVRS** [comment]
Good questions. I'd like the initial zoom level to fill the viewport (maybe with a reasonable amount of padding) and have the max zoom be maybe 2x the viewport size.

Increasing maxZoomPixelRatio might work for me... in this case the only slides are similar SVG images.

I would look into adjusting the width and height specified in the file itself, but I get the file in the form of a blob from an API response and this happens at scale.

**Reply by igordanchenko** [reply]
> I'd like the initial zoom level to fill the viewport (maybe with a reasonable amount of padding) 

Can you see if the following render function works the way you expect?

```jsx
<Lightbox
  render={{
    slide: ({ slide }) => {
      if (isImageSlide(slide) && slide.src.endsWith(".svg")) {
        return (
          <img
            alt=""
            draggable="false"
            className="yarl__slide_image"
            src={slide.src}
            width="100%"
            height="100%"
          />
        );
      }
      return undefined;
    },
  }}
  // ...
/>
```

> and have the max zoom be maybe 2x the viewport size

You can provide `width` and `height` attributes in the `photos` array:

```jsx
slides={[{ src: "./image.svg", width: 4000, height: 4000 }]}
```

**Reply by JPVRS** [reply]
I get an error:

> ReferenceError: isImageSlide is not define

This is what I'm testing with:

```
<Lightbox
  render={{
    slide: ({ slide, rect }) => {
      if (isImageSlide(slide) && slide.src.endsWith(".svg")) {
        return (
          <img
            alt=""
            draggable="false"
            className="yarl__slide_image"
            src={slide.src}
            width="100%"
            height="100%"
          />
        );
      }
      return undefined;
    },
    buttonPrev: () => null,
    buttonNext: () => null,
  }}
  open={isLightBoxOpen}
  close={() => setIsLightBoxOpen(false)}
  plugins={[Zoom]}
  zoom={{ ref: zoomRef }}
  slides={[
    {
      src: dataElement.imageBlobUrl,
      width: 4000,
      height: 4000,
    },
  ]}
  styles={{
    container: { backgroundColor: "rgba(255, 255, 255)" },
  }}
/>
```

Part of the issue might be that the SVG is in the form of a "blob:" URL and not a raw SVG file.

**Reply by igordanchenko** [reply]
> ReferenceError: isImageSlide is not define

You need to import it from the package.

```js
import Lightbox, { isImageSlide } from "yet-another-react-lightbox";
```

**Reply by JPVRS** [reply]
Importing like you mentioned resolved that error. New error (which makes sense given the blob URL):

```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'endsWith')

  227 | render={{
  228 |   slide: ({ slide, rect }) => {
> 229 |     if (isImageSlide(slide) && slide.src.endsWith(".svg")) {
      |                                         ^
  230 |       return (
  231 |         <img
  232 |           alt=""
```

The image URL is like this:

blob:http://localhost:3000/571ac659-381a-49e0-8820-ff3ddb2b5afa

So there's no "svg" at the end for the render function to detect.

I'm happy to spend a moment creating a minimal repro if you can let me know the best way to do that. I'll admit that I'm relatively new at discussing/demonstrating these types of issues.

**Reply by igordanchenko** [reply]
> Part of the issue might be that the SVG is in the form of a "blob:" URL and not a raw SVG file.

You may need to update the `if` clause to match your use case. For example, `if (isImageSlide(slide) && slide.src.startsWith("blob:"))`

**Reply by JPVRS** [reply]
That gives me a different error, one that's odd considering that "copy image URL" gives me a URL starting with "blob:"

```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'startsWith')

  229 | if (
  230 |   isImageSlide(slide) &&
> 231 |   slide.src.startsWith("blob:")
      |            ^
  232 | ) {
  233 |   return (
  234 |     <img
```

**Reply by igordanchenko** [reply]
That's odd... Can you add optional chaining:

```
slide.src?.startsWith("blob:")
```

**Reply by JPVRS** [reply]
The error goes away, but the Lightbox is empty.

**Reply by igordanchenko** [reply]
I guess it's time to see a repro. Can you clone the following sandbox and update it with your specific image to reproduce the issue?

https://codesandbox.io/p/devbox/yet-another-react-lightbox-sandbox-mgc04i

**Comment by JPVRS** [comment, marked as answer]
I seem to have found an acceptable solution by resizing the SVG programmatically as a percentage of the viewport width before passing it to the Lightbox.

**Comment by JPVRS** [comment]
Just following up on this to say that the reason I didn't create the repro was that I couldn't figure out an easy way to add the SVG blob part of the problem into the codesandbox. Since it's really easy to resize the SVG based on the viewport before passing it to the lightbox, I personally would consider this solved (in agreement with that being marked as the answer).
